src/main/scala/chiseltest/backends/SimulatorInterface.scala

```diff
@@ -0,0 +1,30 @@
+package chiseltest.backends
+
+import firrtl.AnnotationSeq
+
+/** API to Simulator. */
+trait SimulatorInterface {
+
+  /** start time of this test. */
+  final val startTime: Long = System.nanoTime()
+
+  /** how many nano seconds has elapsed since this test start. */
+  def elapsedNanoSeconds(): Long = System.nanoTime() - startTime
+
+  def annotations: AnnotationSeq
+
+  /** poke a signal name with a [[BigInt]]. */
```

Comment:
Consider defining behavior if the signal doesn't exist - this could be an exception or a return code. I think the exception makes more sense, so things fail noisily?

Reply:
Also, define the naming convention (this can refer to some other documentation / ScalaDoc) and/or leave a TODO if this is to be refactored to a non-Stringly-typed interface in the future.